contracts/factory/src/tests/queries.rs

```diff
@@ -1399,3 +1403,171 @@ fn test_query_user_portfolio_without_stake() {
         }
     );
 }
+
+#[test]
+fn test_ttl_extensions_with_multiple_pool_queries() {
+    let env = Env::default();
+    env.mock_all_auths();
+    env.cost_estimate().budget().reset_unlimited();
+
+    env.ledger().with_mut(|li| {
+        li.min_persistent_entry_ttl = DAY_IN_LEDGERS; // 17_280
+        li.max_entry_ttl = 30 * DAY_IN_LEDGERS; // 518_400
+    });
+
+    let admin = Address::generate(&env);
+    let fee_recipient = Address::generate(&env);
+    let manager = Address::generate(&env);
+
+    let mut token1 = token_contract::Client::new(
+        &env,
+        &env.register_stellar_asset_contract_v2(admin.clone())
+            .address(),
+    );
+    let mut token2 = token_contract::Client::new(
+        &env,
+        &env.register_stellar_asset_contract_v2(admin.clone())
+            .address(),
+    );
+
+    if token2.address < token1.address {
+        std::mem::swap(&mut token1, &mut token2);
+    }
+
+    let factory = deploy_factory_contract(&env, Some(admin.clone()));
+    let factory_address = factory.address.clone();
+    let lp_init_info = generate_lp_init_info(
+        token1.address.clone(),
+        token2.address.clone(),
+        manager,
+        admin.clone(),
+        fee_recipient,
+    );
+
+    let lp_address = factory.create_liquidity_pool(
+        &admin,
+        &lp_init_info,
+        &String::from_str(&env, "Pool"),
+        &String::from_str(&env, "PHO/BTC"),
+        &PoolType::Xyk,
+        &None::<u64>,
+        &100i64,
+        &1_000,
+    );
+
+    // initial TTL verification
+    let (initial_instance_ttl, initial_persistent_ttl) = env.as_contract(&factory_address, || {
+        (
+            env.storage().instance().get_ttl(),
+            env.storage().persistent().get_ttl(&DataKey::LpVec),
+        )
+    });
+    // validate initial state
+    assert_eq!(initial_instance_ttl, INSTANCE_BUMP_AMOUNT);
+    assert!(
+        initial_persistent_ttl >= PERSISTENT_BUMP_AMOUNT - 1,
+        "Initial persistent TTL should be at least {} (was {})",
+        PERSISTENT_BUMP_AMOUNT - 1,
+        initial_persistent_ttl
+    );
+
+    // first extension
+    env.ledger().with_mut(|li| {
+        li.sequence_number += INSTANCE_BUMP_AMOUNT - INSTANCE_LIFETIME_THRESHOLD + 1;
+    });
+
+    let _ = factory.query_pools();
+    let (instance_ttl1, persistent_ttl1) = env.as_contract(&factory_address, || {
+        (
+            env.storage().instance().get_ttl(),
+            env.storage().persistent().get_ttl(&DataKey::LpVec),
+        )
+    });
+
+    assert_eq!(instance_ttl1, INSTANCE_BUMP_AMOUNT);
+    assert!(
+        persistent_ttl1 >= PERSISTENT_BUMP_AMOUNT - 1,
+        "Persistent TTL after first extension should be at least {} (was {})",
+        PERSISTENT_BUMP_AMOUNT - 1,
+        persistent_ttl1
+    );
```

Comment:
And this, how is it making sense? 
Why is instance TTL equal only to the amount we supposed to bump it with?
Why are you checking only that persistent TTL is bigger than the bump amount? Why not comparing it with the previous one and to show that the difference between them is equal to that bump amount?